runtime/oti/VMHelpers.hpp

```diff
@@ -2070,57 +2079,76 @@ class VM_VMHelpers
 	static VMINLINE bool
 	isContinuationMounted(J9VMContinuation *continuation)
 	{
-		return J9_ARE_ANY_BITS_SET(continuation->state, ~(uintptr_t)J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN);
+		return J9_ARE_ANY_BITS_SET(continuation->state, ~(uintptr_t)J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_ALL);
 	}
 
 	static VMINLINE bool
-	isContinuationMountedOrConcurrentlyScanned(J9VMContinuation *continuation)
+	isContinuationMountedOrConcurrentlyScanned(J9VMContinuation *continuation, uintptr_t checkConcurrentState)
 	{
-		return isContinuationMounted(continuation) || isConcurrentlyScannedFromContinuationState(continuation->state);
+		return isContinuationMounted(continuation) || isConcurrentlyScannedFromContinuationState(continuation->state, checkConcurrentState);
 	}
 
 /*
- *	If low tagging failed due to either
+ *
+ * param[in] checkConcurrentState can be J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_LOCAL or J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_GLOBAL
+ *
+ *  If WinningConcurrentGCScan set J9_GC_CONTINUATION_bit0:STATE_CONCURRENT_SCAN_LOCAL or bit1:J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_GLOBAL in the state base on checkConcurrentState
+ *	If low tagging(bit0 or bit1) failed due to either
  *
  *   a carrier thread winning to mount, we don't need to do anything, since it will be compensated by pre/post mount actions
- *   another GC thread winning to scan, again don't do anything, and let the winning thread do the work, instead
+ *   another GC thread winning to scan(bit0/bit1,bit0 and bit1 is irrelevant and independent), again don't do anything, and let the winning thread do the work, instead
  */
 	static VMINLINE bool
-	tryWinningConcurrentGCScan(J9VMContinuation *continuation)
+	tryWinningConcurrentGCScan(J9VMContinuation *continuation, uintptr_t checkConcurrentState)
 	{
-		return J9_GC_CONTINUATION_STATE_INITIAL == VM_AtomicSupport::lockCompareExchange(&continuation->state, J9_GC_CONTINUATION_STATE_INITIAL, J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN);
+		uintptr_t state;
+		uintptr_t retState;
+		do {
+			state = continuation->state & (J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_ALL & (~checkConcurrentState));
```

Comment:
Let's introduce a small helper, that could help with easier code understanding. Could be used here twice, and in 'exit' method, too. 


/**
 * extract concurrent state of local GC, if this GC is global, and vice-versa
 * extract concurrent state of global GC, if this GC is local
 */

UDATA complementGCConcurrentState(UDATA continuationState, UDATA thisGCConcurrentState) {

        return continuationState & (J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_ALL & ~thisGCConcurentState);
}